stdlib/ballerina-config/src/main/java/org/ballerinalang/bcl/parser/BConfigLangListener.java

```diff
@@ -58,8 +64,18 @@ public void exitKeyVal(TomlParser.KeyValContext context) {
     }
 
     @Override
-    public void enterKey(TomlParser.KeyContext context) {
-        currentKey = context.getText();
+    public void enterDotSep(TomlParser.DotSepContext context) {
+        currentKey.append(context.getText());
+    }
+
+    @Override
+    public void enterQuotedKey(TomlParser.QuotedKeyContext context) {
+        currentKey = currentKey.append(new StringBuilder(context.basicString().basicStringValue().getText()));
```

Comment:
Why do we need a new `StringBuilder` here? Simply appending the basicStringValue would do right?

Reply:
Yes, this is a mistake.